lib/Error.js

```diff
@@ -1,142 +1,138 @@
-'use strict';
-
+"use strict";
 /**
  * StripeError is the base error from which all other more specific Stripe errors derive.
  * Specifically for errors returned from Stripe's REST API.
  */
 class StripeError extends Error {
-  constructor(raw = {}) {
-    super(raw.message);
-    this.type = this.constructor.name;
-
-    this.raw = raw;
-    this.rawType = raw.type;
-    this.code = raw.code;
-    this.doc_url = raw.doc_url;
-    this.param = raw.param;
-    this.detail = raw.detail;
-    this.headers = raw.headers;
-    this.requestId = raw.requestId;
-    this.statusCode = raw.statusCode;
-    this.message = raw.message;
-
-    this.charge = raw.charge;
-    this.decline_code = raw.decline_code;
-    this.payment_intent = raw.payment_intent;
-    this.payment_method = raw.payment_method;
-    this.payment_method_type = raw.payment_method_type;
-    this.setup_intent = raw.setup_intent;
-    this.source = raw.source;
-  }
-
-  /**
-   * Helper factory which takes raw stripe errors and outputs wrapping instances
-   */
-  static generate(rawStripeError) {
-    switch (rawStripeError.type) {
-      case 'card_error':
-        return new StripeCardError(rawStripeError);
-      case 'invalid_request_error':
-        return new StripeInvalidRequestError(rawStripeError);
-      case 'api_error':
-        return new StripeAPIError(rawStripeError);
-      case 'authentication_error':
-        return new StripeAuthenticationError(rawStripeError);
-      case 'rate_limit_error':
-        return new StripeRateLimitError(rawStripeError);
-      case 'idempotency_error':
-        return new StripeIdempotencyError(rawStripeError);
-      case 'invalid_grant':
-        return new StripeInvalidGrantError(rawStripeError);
-      default:
-        return new StripeUnknownError(rawStripeError);
+    constructor(raw) {
```

Comment:
This is a little different. Please add default value to `raw`. Currently, the lines like `this.code = raw.code;` will fail with `Cannot read properties of undefined`.